src/h2load.cc

```diff
@@ -1744,42 +1743,28 @@ int main(int argc, char **argv) {
               << "cores." << std::endl;
   }
 
-  if (!config.is_rate_mode()) {
-    if (config.nreqs < config.nclients) {
-      std::cerr << "-n, -c: the number of requests must be greater than or "
-                << "equal to the concurrent clients." << std::endl;
-      exit(EXIT_FAILURE);
-    }
+  if (config.nreqs < config.nclients) {
+    std::cerr << "-n, -c: the number of requests must be greater than or "
+              << "equal to the concurrent clients." << std::endl;
+    exit(EXIT_FAILURE);
+  }
 
-    if (config.nclients < config.nthreads) {
-      std::cerr << "-c, -t: the number of client must be greater than or equal "
-                   "to the number of threads." << std::endl;
-      exit(EXIT_FAILURE);
-    }
-  } else {
+  if (config.nclients < config.nthreads) {
+    std::cerr << "-c, -t: the number of client must be greater than or equal "
+                 "to the number of threads." << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  if (config.is_rate_mode()) {
     if (config.rate < config.nthreads) {
       std::cerr << "-r, -t: the connection rate must be greater than or equal "
```

Comment:
This restriction is still in place, with the code refactor are the calculations simplified to remove or reduce it?

For instance it would be nice to have something like `h2load -t6 -c360 -r1`, where each thread creates 60 clients at a rate of 1 client/second. This gives a ramp-up period of 60 seconds. With `-r6` the ramp up period is 10 seconds.

Its not all bad news though as a workaround at the moment is `h2load -t6 -c300 -r6 --rate-period=10`, with gives a ramp-up period of 60 seconds (as above) but with bigger steps.